### ISSUE TITLE
flannel does not pick latest value from etcd

### ISSUE THREAD

**Issue by gouyang** [coreos/flannel#196]
Set /coreos.com/network/config and run flanneld, change  /coreos.com/network/config to different values, run flanneld again, it still use the old values.

$ ./bin/etcdctl get /coreos.com/network/config
{
  "Network": "18.17.0.0/16",
  "SubnetLen": 24,
  "Backend": {
    "Type": "udp",
    "VNI": 1
  }
}

$ vi flannel-config.json 
$ curl -L http://localhost:4001/v2/keys/coreos.com/network/config -XPUT --data-urlencode value@flannel-config.json 
{"action":"set","node":{"key":"/coreos.com/network/config","value":"{\n  \"Network\": \"18.16.0.0/16\",\n  \"SubnetLen\": 24,\n  \"Backend\": {\n    \"Type\": \"udp\",\n    \"VNI\": 1\n  }\n}\n","modifiedIndex":12,"createdIndex":12},"prevNode":{"key":"/coreos.com/network/config","value":"{\n  \"Network\": \"18.17.0.0/16\",\n  \"SubnetLen\": 24,\n  \"Backend\": {\n    \"Type\": \"udp\",\n    \"VNI\": 1\n  }\n}\n","modifiedIndex":5,"createdIndex":5}}

$ ./bin/etcdctl get /coreos.com/network/config
{
  "Network": "18.16.0.0/16",
  "SubnetLen": 24,
  "Backend": {
    "Type": "udp",
    "VNI": 1
  }
}

$ sudo ./bin/flanneld -etcd-endpoints=http://localhost:4001
I0602 15:12:54.546376 24636 main.go:275] Installing signal handlers
I0602 15:12:54.546500 24636 main.go:141] Determining IP address of default interface
I0602 15:12:54.546943 24636 main.go:189] Using 10.66.11.39 as external interface
go-etcd2015/06/02 15:12:54 DEBUG: get /coreos.com/network/config [http://localhost:4001]
go-etcd2015/06/02 15:12:54 DEBUG: Connecting to etcd: attempt 1 for keys/coreos.com/network/config?quorum=false&recursive=false&sorted=false
go-etcd2015/06/02 15:12:54 DEBUG: send.request.to http://localhost:4001/v2/keys/coreos.com/network/config?quorum=false&recursive=false&sorted=false | method GET
go-etcd2015/06/02 15:12:54 DEBUG: recv.response.from 
go-etcd2015/06/02 15:12:54 DEBUG: recv.success 
go-etcd2015/06/02 15:12:54 DEBUG: get /coreos.com/network/config [http://localhost:4001]
go-etcd2015/06/02 15:12:54 DEBUG: Connecting to etcd: attempt 1 for keys/coreos.com/network/config?quorum=false&recursive=false&sorted=false
go-etcd2015/06/02 15:12:54 DEBUG: send.request.to http://localhost:4001/v2/keys/coreos.com/network/config?quorum=false&recursive=false&sorted=false | method GET
go-etcd2015/06/02 15:12:54 DEBUG: recv.response.from 
go-etcd2015/06/02 15:12:54 DEBUG: recv.success 
go-etcd2015/06/02 15:12:54 DEBUG: get /coreos.com/network/subnets [http://localhost:4001]
go-etcd2015/06/02 15:12:54 DEBUG: Connecting to etcd: attempt 1 for keys/coreos.com/network/subnets?quorum=false&recursive=true&sorted=false
go-etcd2015/06/02 15:12:54 DEBUG: send.request.to http://localhost:4001/v2/keys/coreos.com/network/subnets?quorum=false&recursive=true&sorted=false | method GET
go-etcd2015/06/02 15:12:54 DEBUG: recv.response.from 
go-etcd2015/06/02 15:12:54 DEBUG: recv.success 
go-etcd2015/06/02 15:12:54 DEBUG: put /coreos.com/network/subnets/18.17.84.0-24, {"PublicIP":"10.66.11.39"}, ttl: 86400, [http://localhost:4001]
go-etcd2015/06/02 15:12:54 DEBUG: Connecting to etcd: attempt 1 for keys/coreos.com/network/subnets/18.17.84.0-24
go-etcd2015/06/02 15:12:54 DEBUG: send.request.to http://localhost:4001/v2/keys/coreos.com/network/subnets/18.17.84.0-24 | method PUT
go-etcd2015/06/02 15:12:54 DEBUG: recv.response.from 
go-etcd2015/06/02 15:12:54 DEBUG: recv.success 
I0602 15:12:54.604657 24636 etcd.go:91] Subnet lease acquired: 18.17.84.0/24
I0602 15:12:54.606004 24636 udp.go:222] Watching for new subnet leases
go-etcd2015/06/02 15:12:54 DEBUG: get /coreos.com/network/subnets [http://localhost:4001]
go-etcd2015/06/02 15:12:54 DEBUG: Connecting to etcd: attempt 1 for keys/coreos.com/network/subnets?quorum=false&recursive=true&sorted=false
go-etcd2015/06/02 15:12:54 DEBUG: send.request.to http://localhost:4001/v2/keys/coreos.com/network/subnets?quorum=false&recursive=true&sorted=false | method GET
go-etcd2015/06/02 15:12:54 DEBUG: recv.response.from 
go-etcd2015/06/02 15:12:54 DEBUG: recv.success 
go-etcd2015/06/02 15:12:54 DEBUG: rawWatch /coreos.com/network/subnets []
go-etcd2015/06/02 15:12:54 DEBUG: get /coreos.com/network/subnets [http://localhost:4001]
go-etcd2015/06/02 15:12:54 DEBUG: Connecting to etcd: attempt 1 for keys/coreos.com/network/subnets?recursive=true&wait=true&waitIndex=14
go-etcd2015/06/02 15:12:54 DEBUG: send.request.to http://localhost:4001/v2/keys/coreos.com/network/subnets?recursive=true&wait=true&waitIndex=14 | method GET
I0602 15:12:54.607151 24636 udp.go:247] Subnet added: 18.17.84.0/24
go-etcd2015/06/02 15:12:54 DEBUG: recv.response.from

**Comment by eyakubovich** [comment]
I think the reason this happens is because the old lease has not yet expired and gets reused instead of acquiring a new one. However, this is functionally a bug and I'm marking it as such.

**Comment by akamalov** [comment]
I am seconding this behavior. Changed the network segment to something other than it was in ```/coreos.com/network/config```, restarted ```flanneld``` and it is still using the old network segment. I am using  ```CoreOS 695.0.0```. 

Display changed values in ```/coreos.com/network/config```:

```
wwserver60 bin # etcdctl get /coreos.com/network/config
{ "Network": "10.100.0.0/16","SubnetLen":20,"SubnetMin":"10.100.10.0","SubnetMax":"10.100.99.0","Backend":{"Type":"vxlan","Port":7890 }}
wwserver60 bin # 
```

Note: I am moving network segment from ```11.0.0.0/8``` to ```10.100.0.0/16```.

- Restarted ```flanneld```

- Displaying interface on ```flannel.1``` interface:

```
wwserver60 bin # ifconfig flannel.1
flannel.1: flags=4163<UP,BROADCAST,RUNNING,MULTICAST>  mtu 1450
        inet 11.13.64.0  netmask 255.255.0.0  broadcast 0.0.0.0
        inet6 fe80::dc32:82ff:fedb:3bf8  prefixlen 64  scopeid 0x20<link>
        ether de:32:82:db:3b:f8  txqueuelen 0  (Ethernet)
        RX packets 0  bytes 0 (0.0 B)
        RX errors 0  dropped 0  overruns 0  frame 0
        TX packets 0  bytes 0 (0.0 B)
        TX errors 0  dropped 49 overruns 0  carrier 0  collisions 0

wwserver60 bin # 
```

As you can see, interface ```flannel.1``` is retaining old network segment IP.

Is it possible to expedite this ?

Thanks a ton.

Alex

**Comment by eyakubovich** [comment]
We will do our best to get this fixed before the next release.

**Comment by akamalov** [comment]
Thanks Eugene. I am deploying k8s environment on top of CoreOS, and right now I am dead in the water without this fix.

Thanks again,

Alex

**Comment by akamalov** [comment]
Eugene, does it mean the fix is coming in next release?

Thanks,

Alex

On Mon, Jun 22, 2015 at 8:31 PM, Eugene Yakubovich <notifications@github.com

**Comment by eyakubovich** [comment]
@akamalov Yes, the fix will be included in the next release -- which I'm shooting for by the end of this week. However, that will be a release of flannel, it will be a bit more before it appears in CoreOS. But you'll be able to use a drop-in with `FLANNEL_VER` to point it to the new version.

**Comment by suyogbarve** [comment]
Is there a temporary work-around to get the new ip range?

**Comment by MohdAhmad** [comment]
@suyogbarve this was fixed in the [flannel v0.5](https://github.com/coreos/flannel/releases/tag/v0.5.0) release. We have had a maintenance, [flannel v0.51] (https://github.com/coreos/flannel/releases/tag/v0.5.1) release out since then as well.
